### DISCUSSION TITLE
Maddalena Arrow Evo 868 - A2 frame data (manufacturer-specific encoded frame)

### DISCUSSION THREAD

**Discussion by matevzsi** [wmbusmeters/wmbusmeters#749] [Ideas]
Has anybody come accross the encrypted data of Maddalena Arrow Evo 868 water meter? It uses T1 frames with A2 (manufacturer specific) data frame, which is partly encoded.

(Un)fortunately, we have this water meter installed by our utility company.  I was able to record the data using the HackRF and use it to find the cipher, which changes twice per year. I was not able to determine how (and if) the ciper is related to the serial number and/or measurement period. Therefore, I need to update the cipher manually twice per year to continue gathering meter data.
Lately, I was able to port the code to the Adafruit Feather 32u4 Radio with RFM69HCW, which has better sensitivity than the SDR-based solution and is much less CPU-heavy... The code if far from being clean, but it works for now...

My repository if anybody is willing to take a look at the data coding - https://github.com/matevzsi/WaterMeterReader_ArrowEvo868

**Comment by nikito7** [comment]
Any update?

**Comment by matevzsi** [comment]
I had it working for a year with some periodic manual code decryption steps, then our water utility company replaced them with a completely different type of meter and I went to https://github.com/jomjol/AI-on-the-edge-device since the new meters have limited battery life for readouts.

**Reply by nikito7** [reply]
It's just like Izar, with some changes.

But mine it's rebranded just say "wmbus arrow".

https://github.com/wmbusmeters/wmbusmeters/issues/1416
